test_runner/regress/test_compaction.py

```diff
@@ -320,3 +332,93 @@ def assert_broken():
         or 0
     ) == 0
     assert not env.pageserver.log_contains(".*Circuit breaker failure ended.*")
+
+
+@pytest.mark.parametrize("enabled", [True, False])
+def test_image_layer_compression(neon_env_builder: NeonEnvBuilder, enabled: bool):
+    tenant_conf = {
+        # small checkpointing and compaction targets to ensure we generate many upload operations
+        "checkpoint_distance": f"{128 * 1024}",
+        "compaction_threshold": "1",
+        "compaction_target_size": f"{128 * 1024}",
```

Comment:
As for writing new tests which such a small checkpoint_distance, I would really like to avoid that because it triggers the initdb image layer optimization as the flush happens *after* importing the images *before* ingesting the initdb wal.

Are these actually even needed by the test? Layer count might not matter that much.

Reply:
I don't want the test to just use one layer file, ideally we are closer to a scenario where there is many layer files, as that is closer to prod.

Reply:
Ok. At least this test now exercises all image generation paths.